code/obj/item/staffkiosk.dm

```diff
@@ -0,0 +1,287 @@
+			return
+		else
+			..()
+
+
+/proc/build_recruitment_role_cache()
+	recruitment_roles.Cut()
+	for(var/S in concrete_typesof(/datum/recruitment_role))
+		recruitment_roles += new S()
+
+//datum? I hardly know em
+
+ABSTRACT_TYPE(/datum/recruitment_role)
+/datum/recruitment_role
+	var/name = "Clown Assistant" //title as it will appear on assistant's card
+	var/accessParent = "Clown" //job whose access this mimics
+	var/canAuthorize = list("Clown") //who is allowed to give out this job
+	var/cardIcon = "id_clown" //what icon state the card should receive
+	var/tokenIcon = "assist-token"
+	var/dispensedKit = /obj/item/storage/box/staffkit/med //what kit should be dispensed (lo and behold)
+
+/datum/recruitment_role/medical
+	name = "Medical Assistant"
+	accessParent = "Medical Doctor"
+	canAuthorize = list("Medical Doctor","Geneticist","Roboticist","Medical Director","Head of Personnel","Captain")
+	cardIcon = "id_res"
+	tokenIcon = "assist-token-med"
+	dispensedKit = /obj/item/storage/box/staffkit/med
+
+/datum/recruitment_role/robo
+	name = "Robotics Assistant"
+	accessParent = "Roboticist"
+	canAuthorize = list("Roboticist","Medical Director","Head of Personnel","Captain")
+	cardIcon = "id_res"
+	tokenIcon = "assist-token-robo"
+	dispensedKit = /obj/item/storage/box/staffkit/robo
+
+/datum/recruitment_role/engineer
+	name = "Engineering Assistant"
+	accessParent = "Engineer"
+	canAuthorize = list("Engineer","Chief Engineer","Head of Personnel","Captain")
+	cardIcon = "id_eng"
+	tokenIcon = "assist-token-eng"
+	dispensedKit = /obj/item/storage/box/staffkit/eng
+
+/datum/recruitment_role/mechanic
+	name = "Mechanics Assistant"
+	accessParent = "Mechanic"
+	canAuthorize = list("Mechanic","Chief Engineer","Head of Personnel","Captain")
+	cardIcon = "id_eng"
+	tokenIcon = "assist-token-mech"
+	dispensedKit = /obj/item/storage/box/staffkit/mech
+
+/datum/recruitment_role/mining
+	name = "Mining Assistant"
+	accessParent = "Miner"
+	canAuthorize = list("Miner","Chief Engineer","Head of Personnel","Captain")
+	cardIcon = "id_eng"
+	tokenIcon = "assist-token-mine"
+	dispensedKit = /obj/item/storage/box/staffkit/miner
+
+/datum/recruitment_role/sci
+	name = "Research Assistant"
+	accessParent = "Scientist"
+	canAuthorize = list("Scientist","Research Director","Head of Personnel","Captain")
+	cardIcon = "id_res"
+	tokenIcon = "assist-token-sci"
+	dispensedKit = /obj/item/storage/box/staffkit/res
+
+/datum/recruitment_role/chef
+	name = "Culinary Assistant"
+	accessParent = "Chef"
+	canAuthorize = list("Chef","Head of Personnel","Captain")
+	cardIcon = "id_civ"
+	tokenIcon = "assist-token-chef"
+	dispensedKit = /obj/item/storage/box/staffkit/chef
+
+/datum/recruitment_role/bar
+	name = "Bar Assistant"
+	accessParent = "Bartender"
+	canAuthorize = list("Bartender","Head of Personnel","Captain")
+	cardIcon = "id_civ"
+	tokenIcon = "assist-token-bar"
+	dispensedKit = /obj/item/storage/box/staffkit/bar
+
```

Comment:
No Janitor?!?!? 🧽🧼🧹 😱

Reply:
Not initially, no. Seems a slightly weird role to be recruiting assistants for, but a janitorial assistant could be added down the line if there's interest in it.